src/Parser/AfflictionWarlock/Modules/SoulShards/SoulShardBreakdown.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+class SoulShardBreakdown extends React.Component {
+  static propTypes = {
+    shardsGained: PropTypes.object.isRequired,
+    shardsSpent: PropTypes.object.isRequired,
+    shardsWasted: PropTypes.object.isRequired,
+  };
+  prepareGenerated(shardGen, shardWasted){
+    //shardGen and shardWasted has the same number of abilities (some having 0, these can be now filtered)
+    const abilities = [];
+    Object.keys(shardGen).map(abilityId => abilities.push({abilityId: Number(abilityId), generated: shardGen[abilityId].shards, wasted: shardWasted[abilityId].shards}));
+    abilities.sort((a, b) => b.generated - a.generated);
+    return abilities;
```

Comment:
This can be written like:

```es6
return Object.keys(shardGen)
  .map(abilityId => ({
    abilityId: Number(abilityId),
    generated: shardGen[abilityId].shards,
    wasted: shardWasted[abilityId].shards,
  }))
  .sort((a, b) => b.generated - a.generated);
```

The pushing in a map is very weird and should definitely be removed.